### PR TITLE
dasllama-server serves a whole-model GPU slot from the device: K/V regions, device-home streams, and a page that says how a slot is served

### PR DESCRIPTION
**Pull request by borisbat**
**Behavior change: `gpu = vulkan` in dasllama-server serves a model that fits the card whole from the GPU, every stream's K/V cache included.**

**Why.** GPU slots served their streams through paged host sessions on the CPU: gemma-4-E2B Q4_K_M read tg128 34 tok/s on an RTX 5060 Ti.

**What changes.**
- The Vulkan whole-model driver holds a K/V region per stream; a device-home session lives in one and has no host cache.
- A finished stream parks its rows; the next turn adopts them.
- `/v1/stats` and the model card say how a slot is served and count calls the GPU handed to the CPU.
- `REVIEW_GPU.md`: a served GPU path that falls back to the CPU is a defect.

**Observable behavior.**
- One chat stream: 31 -> 150 tok/s; bench pp512 141 -> 7915, tg128 34 -> 151 tok/s.
- Four chats: queued -> decoding together, 141 tok/s summed.
- Model card: backend chip -> served line, reason, CPU-call counter.
- A media request under device mode: a driver panic -> refused at `submit`.

**Where to look.** Region ownership in `dasllama_gpu_resident.das` and the scheduler's device mode; `test_gpu_resident_regions*.das` hold one session against another's rows bit for bit.

<details>
<summary>Validation, claims, ledger</summary>

### Validation
- Every number above: `daslang -jit utils/dasllama-server/main.das` under the watchdog, `backend = "gpu"`, `streams = 4`, kv f16, the vulkan tier in its default cm2 mode, the box's tuned sidecar, `DAS_JOBQUE_THREADS=16`, no other override; Threadripper 3990X + RTX 5060 Ti 16 GB, Windows; `gemma-4-E2B-it-Q4_K_M.gguf`. Chat rates are 128-token requests at temperature 0, completion tokens over request wall; bench rows are the server's in-process pp512/tg128, three reps. The one-region / four-region pair with footprints is the `PERF_LEDGER.md` row. No board row exists for a Vulkan path (the record rig has no Vulkan leg) and no listed instrument reaches multi-stream device-home serving; the server's bench route and chat requests are the instruments.
- The GPU cells run only on a box with the card and the models, no CI lane has either. At the final code, `dastest -jit` with `DASLLAMA_GPU=1`: `test_gpu_resident_regions.das` 24 of 24 (Qwen2.5-0.5B), `_hybrid` 11 of 11 (Qwen3.5-0.8B), `_e2b` 9 of 9 (gemma-4-E2B Q8_0). The GPU-vs-CPU parity instruments ran at commit `be824c29b`, before the harvest and audit batches (comment, naming and reporting changes on the driver): `test_gpu_resident_gemma3_1b` 9, `_qwen2` 3, `_gemma4_e2b` 5, `_hybrid` 15, `test_gpu_slot_swap` 2, `test_gpu_model_swap` 6; the K-quant class rides `_hybrid`'s Q4_K_M twins, the MXFP4 carrier (`test_gpu_resident_gptoss.das`) and the f32 mirror codec (`DASLLAMA_VK_KV32=1`) were not run.
- The stocked suite was started at the final code and stopped by decision after 7 files, all green (`test_attn_span`, `test_chat`, `test_deltanet` among them): it runs whole before PR1, with the batched dispatch. The model-free suite at the final code: 75 files, 0 failed, 5 cells skipped (an absent fixture or device, each loudly). make-pr's sync, stamp-reach and jit-smoke gates green; ast-verify clean over the 27 changed `.das` files; the review-md gate green; the dupes report's exact clusters are the tier's pre-existing one-line seat shells, ruled structural.
- `test_openai_server.das` needs live models and a spawned server, so its CI row is compile-only; on this box 24 of 24 at the final server code (the served fields and the no-warning line asserted on the CPU slot). `test_slot_served.das` 8 of 8 interpreted, on the lane's run row. The page suite 158 of 158; master's `models-panel.spec.js` against the re-captured fixtures goes red on the one spec the re-capture moved.
- Negative controls run and restored: the region pin, `pos0` in the window chain, the deltanet flush, the per-row PLE side input, the resident ask, the device prefill chunk floor, the prefill-pin bypass - each turns its cell red.
- The stats fixtures were re-captured from real boots on this box per `fixtures/README.md` (roster, speech, failed speech, setup, and a roster boot after one embeddings request for `stats_cpu_pass.json`); only the failed pack's path was normalized. `smol` reads `flavor: planar` in the roster pair and `vulkan` in the speech capture because the first boot mints the image and a later boot maps it.
- cm2 arm only, one vendor: the diff changes no kernel and no dispatch class (a region is an element offset folded into an existing push constant).

### Claims - stated, not tested
- The parked token list is cut to the rows a region holds (`park_stream`): removing the cut keeps the suite green, because `gpu_device_kv_adopt` refuses more rows than a region holds - a missing cut loses an adoption, it hands out no wrong row.
- A batched step that finds every region pinned returns false and the device-home guard panics: the guard the third-session cell proves, reached from the batched entry.
- `vk_rdec_prepare` declines a mirror past `maxStorageBufferRange` times the region count: read, not run - no stocked model reaches 4 GiB a side on this card.
- Two of the three "not attempted" reasons (no MoE seats, nothing asked for the driver) are read, not run; the switched-off route is tested.
- The bench's device-home arm was verified live (pp512 7915 tok/s, `gpu_cpu_passes` empty); the server test harness serves on the CPU and cannot reach it.

### Not done
- Ledgered in `followup_vulkan.md` item 69, each counted on the page when it fires: a recurrent model's prefill continuation and its one state slot per layer, media-tower slots and self-speculation stay host-cached, the embeddings route's session is host-cached, the regions share one 4 GiB binding a side, and the batched step dispatches its rows one at a time - PR1's kernel work, with the tg128-at-batch-4 bench row. `followup_metal.md` sec.16 carries the Metal twin.
- Declined folds (dupe audit, TEMPLATABLE): one `feed_logits` / `logits_row` for the six resident rigs, one `resident_rig_ready` gate for the five, one `rdec_region_for` over the decode and prefill bind arms.
- Proposed, not applied: `REVIEW_PLACEMENT.md`'s module-global rule widened to struct fields of `Session` / `Model`; `REVIEW_MEASUREMENT.md`'s scope carve-out and its closed instrument list; `modules/dasLLVM/REVIEW.md`'s suite trigger narrowed to code; three `REVIEW.das` gates (the served strings' banned nouns, `find_daslang`'s three lists, `props.limits.*` reads against `vk_ext_roster`); a `fixtures/README.md` row for the pre-existing `streams_finished.json`.

</details>

### CODE REVIEW

File: modules/dasLLAMA/dasllama/dasllama_gpu_tier.das [hunk 1]
```diff
@@ -1513,6 +1538,37 @@ def gpu_tier_status_() : GpuTierStatus {
...
+//! No call has passed yet: the counts are the armed model's, so a drop and every load zero them.
+def public reset_gpu_cpu_passes() {
+    for (n in g_gpu_cpu_pass_counts) {
+        n = 0l
+    }
+}
+
+def public note_gpu_cpu_pass(reason_slot : int) {
+    g_gpu_cpu_pass_counts[reason_slot]++
+}
```
[the remark this answers is not in the document]
Comment (borisbat) [reply modules/dasLLAMA/dasllama/dasllama_gpu_tier.das:1571]: Accepted, fixed in b5ad2cdd7: `set_gpu_cpu_pass_reasons` panics when the reasons and words lists differ in length, and `note_gpu_cpu_pass` panics on a slot outside the registry. Both are driver mistakes rather than runtime conditions, so a panic that names the counts is the right shape - the tuples `gpu_cpu_passes` answers can no longer desync.

File: utils/dasllama-server/openai_server.das [hunk 2]
```diff
@@ -305,16 +333,73 @@ def private bind_gpu_slot(si : int) {
...
+}
+
+def private sync_slot_device_kv(var s : ModelSlot) {
+    let want = slot_wants_device_kv(s)
+    let was = s.sched.device_kv
+    if (set_device_kv(s.sched, want) != was) {
+        let kind = want ? "device-home sessions: every stream's K/V stays on the GPU, {gpu_device_session_ctx()} positions each" : "host-cached sessions"
+        to_log(LOG_INFO, "dasllama-server: {s.name} admits {kind}\n")
+    }
+}
```
[the remark this answers is not in the document]
Comment (borisbat) [reply utils/dasllama-server/openai_server.das]: Accepted, fixed in b5ad2cdd7: `sync_slot_device_kv` now takes whether the slot holds the GPU tier (`si == g_gpu_owner`) and turns device mode on only then. A slot the tier has moved off keeps host-cached sessions until it holds the card again, so it can never admit a device-home session the driver is not armed for.
